macos/docs/Controls/Button.md

```diff
@@ -0,0 +1,37 @@
+# Button
+
+## Overview
+Buttons are one of the core controls that make an app feel native to the platform it's on. It’s important to respect the platform's paradigms in order to help the user feel at home on macOS and keep the experience quality high.
+
+![ButtonViews.png](.attachments/ButtonViews.png)
+
+## Best Practices
+### Do
+- Make sure the label conveys a clear purpose of the button to the user.
+- Describe the action the button will perform, ideally with a verb. Use concise, specific, self-explanatory labels, usually a single word.
+
+### Don't
+- Place the default focus on a button that performs a destructive actionn. Instead, place the default focus on the button that performs the "safe action" (e.g "Cancel").
+
+## Usage
+```Swift
+// There are 3 styles to choose from, the default being primary filled
+let primaryFilledButton = Button(title: "FluentUI Button", style: .primaryFilled),
+let primaryOutlineButton = Button(title: "FluentUI Button", style: .primaryOutline),
+let borderlessButton = Button(title: "FluentUI Button", style: .borderless)
+```
+
+```Swift
+// You can display an image instead of a title
+let buttonWithImageAndStyle = Button(image: NSImage(named: NSImage.addTemplateName)!, style: .primaryFilled)
+```
+
+Overriding the underlying NSButtonCell will result in udefined behavior.
```

Comment:
I almost prefer something a little less formal like

> The FluentUI Button achieves its styling in part by customizing the underlying NSButtonCell. Any direct modifications to the cell will result in undefined behavior.

Reply:
Except this is not true. The customization is achieved by ignoring the bezel, and modifying the layer. I do nothing to the underlying NSButtonCell. Theoretically, you can do whatever you want and things will probably be fine: I just haven't tested that thoroughly hence undefined behavior.

Also worth noting that for future iterations, I will definitely mess with the NSButtonCell so this guidance will change

Reply:
Side note: I don't like how this sentence lies alone. How would you feel about adding a subheader under usage with something like:
```markdown
### Not recommended
- Overriding NSButtonCell is currently undefined behavior
- Displaying both a title and an image will currently not render properly 
```